addons/website/models/website_visitor.py

```diff
@@ -35,6 +35,7 @@ class WebsiteVisitor(models.Model):
     active = fields.Boolean('Active', default=True)
     website_id = fields.Many2one('website', "Website", readonly=True)
     partner_id = fields.Many2one('res.partner', string="Contact", help="Partner of the last logged in user.")
+    parent_id = fields.Many2one('website.visitor', string="Parent", ondelete='set null', help="Main identity")
```

Comment:
Wild proposal: shouldn't we delete all childs now that we somehow consolidate everything on the main visitor ? That way cleanup is complete when we remove a visitor...

Reply:
I think that would defeat the purpose a bit.

If you switch between 2 computers, then you will have 2 different website.visitor records.
With our current implementation:
- When you are on computer A, we create a website.visitor "visitor1" that will save your activity (livechat, wishlist, ...)
- Then you go to computer B, we create a second website.visitor "visitor2"
- Then you log-in on that second computer -> we consolidate "visitor1" and "visitor2", "visitor2" becomes the main one with all your activity on it, "visitor1" is archived (but *not* deleted)
- Then you go back on computer A, we fetch "visitor1" based on the token in your cookies (the search has "active_test" False so we will find the archived record)
- "visitor1" then becomes the main record, it's switched back to active, consolidated, and you get all your preferences back

If we had deleted "visitor1" instead, then the website would have created a brand new visitor, which seems unnecessary and counter-productive.

So we currently have a back-and-forth implementation VS having to re-create and re-link everything if we delete.

What do you think?
(This task is a bit old so maybe I'm saying wrong things, feel free to correct me).

Reply:
Indeed. Removing those without activity in the cron is sufficient.